### PR TITLE
An augeas helper define to deal with configs

### PR DESCRIPTION
**Pull request by vStone**
Helps with the configuration of a plugin and reloads (or restarts) jenkins after config changes.

### CODE REVIEW

**Comment by jhoblitt** [comment]
@vStone The quality of this code is great!  I need to find the time to think about this feature before merging it unless one of the other maintainers beats me to.

**Comment by vStone** [comment]
Looks like this could use some more improvement. Now I require the cli class to be enabled. We could probably fall back to notifying the `Service` in stead of doing a reload/safe-restart when the cli is not used.

File: spec/defines/jenkins_augeas_spec.rb [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  describe "with plugin param wrong type" do
+    let (:params) {{:config_filename => 'foo.xml', :changes => [], :plugin => ['foo','bar'] }}
+    it do
+      is_expected.to raise_error(Puppet::Error, /is not a string/i)
+    end
+  end
+
+
+  #-------------------------------------------------------------------------------
+  #      |              o                             o
```
Comment (rtyler) [inline spec/defines/jenkins_augeas_spec.rb]: this is silly :P

Comment (vStone) [inline spec/defines/jenkins_augeas_spec.rb]: ¯\_(ツ)_/¯  its only a `git rm` away

**Pull request merged**